Backend/app/database/DatabaseConnectionManager.py

```diff
@@ -41,15 +45,38 @@ def get_collection_connection(cls, collection_name: DatabaseCollection) -> Colle
     def init_database_connection(
         cls, environment: AppEnvironmentMode, connection_uri: str
     ) -> None:
-        """Initializes the database connection and loads its unique instance\
-            based on current environment value
+        """
+        Initializes the database connection.
 
         Args:
-            environment (AppEnvironmentMode): the current environment value
-            connection_uri (str): the database connection uri
+            environment (AppEnvironmentMode): The current environment mode.
+            connection_uri (str): The URI for connecting to the database.
+
+        Returns:
+            None
         """
+        from pymongo import MongoClient
```

Comment:
The purpose of this class is not having any initialization of the Database until we know if tests will be run or we're in production mode. `database_connection_class` contains the logic associated to a concrete database, being a mock connectior or a real one. We're also abstracting the logic behind the `DatabaseConnection` classes, `DatabaseConnectionManager` is only managing that instance and showing it to the rest of the app, concrete logic should be inside `DatabaseConnection` classes :). Feel free to ask if something isn't explained correctly